sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/PartitionPump.cs

```diff
@@ -118,13 +124,24 @@ public async Task StartAsync()
                 {
                     if (RunningTask == null)
                     {
+                        // We expect the token source to be null, but we are playing safe.
+
                         RunningTaskTokenSource?.Cancel();
                         RunningTaskTokenSource = new CancellationTokenSource();
 
                         InnerConsumer = InnerClient.CreateConsumer(ConsumerGroup, PartitionId, Options.InitialEventPosition);
 
+                        // In case an exception is encountered while partition processor is initializing, don't catch it
+                        // and let the event processor handle it.  The inner consumer hasn't connected to the service yet,
+                        // so there's no need to close it.
+
                         await PartitionProcessor.InitializeAsync().ConfigureAwait(false);
 
+                        // Before closing, the running task will set the close reason in case of failure.  When something
+                        // unexpected happens and it's not set, the default value (Unknown) is kept.
+
+                        CloseReason = default;
```

Comment:
No need to do this.  The type initializer will set the same value.

Reply:
Agreed. I'll keep the comment because I think it contains relevant information.